autoflip_calculators/shot_change_fusing/shot_change_fusing_calculator.cc

```diff
@@ -0,0 +1,273 @@
+// Copyright 2019 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "mediapipe/examples/desktop/autoflip/autoflip_messages.pb.h"
+#include "mediapipe/examples/desktop/autoflip/calculators/shot_change_fusing_calculator.pb.h"
+#include "mediapipe/framework/calculator_framework.h"
+#include "mediapipe/framework/port/ret_check.h"
+#include "mediapipe/framework/port/status.h"
+
+using mediapipe::Packet;
+using mediapipe::PacketTypeSet;
+
+constexpr char kIsShotBoundaryTag[] = "SHOT_BOUNDARY";
+constexpr char kOutputTag[] = "OUTPUT";
+const int kInfity = 10000; 
+const int kIdInitial = -6345;
```

Comment:
I think it's better to use -1 (but if there is a reason for -6345 then please comment)

Reply:
At first I used -1. I am just afraid that some users may name the id arbitrarily. Since I use this kIdInitial to check whether the signal is missing. -6345 is actually harder for a user to choose. This is my apartment number.